File: packages/audio/android/src/main/java/dev/tcode/thinmpr/audio/service/contract/SongServiceContract.kt
package dev.tcode.thinmpr.audio.service.contract

interface SongServiceContract {
    fun getAllSongs(): List<Map<String, Any>>

    fun getSongsByArtistId(artistId: String): List<Map<String, Any>> 

    fun getSongsByIds(ids: List<String>): List<Map<String, Any>>

    fun getSongsByAlbumId(albumId: String): List<Map<String, Any>>

    fun getSongById(id: String): Map<String, Any>?
}

File: packages/audio/ios/Service/Contract/SongServiceContract.swift
protocol SongServiceContract {
    func getAllSongs() -> [[String: Any]]

    func getSongsByAlbumId(albumId: String) -> [[String: Any]]

    func getSongsByArtistId(artistId: String) -> [[String: Any]]

    func getSongById(id: String) -> [String: Any]?

    func getSongsByIds(ids: [String]) -> [[String: Any]]
}
